File: golang/008.go
package main

import "fmt"

func findNum(arr []int, target int, start, stop int) int {
	if start  > stop {
		return -1
	}
	mid := start +(stop - start )/ 2

	// fmt.Println(mid)
	

	if arr[mid] == target{
		return mid
	}

	if target > arr[mid]{
		return findNum(arr, target,mid+1,stop)
	} else {
		return findNum(arr, target, start, mid-1 )
	}
}

func main(){
	fmt.Println(findNum([]int{2,3,5,6,7,8},10,0,5))
}

File: python/008.py
def binarySort(arr, target, start, stop):
    if start > stop:
        return -1
    
    midIndex = start + (stop-start)//2

    if target == arr[midIndex]:
        return midIndex

    if target > arr[midIndex]:
        return binarySort(arr, target, midIndex + 1, stop)
    else :
        return binarySort(arr, target, start, midIndex - 1)
    
print(binarySort([2,4,5,6,7,8,10],10,0,6))